src/components/MyVisitForm/MyVisitForm.jsx

```diff
@@ -0,0 +1,138 @@
+import React, { useState, useEffect } from 'react';
+import { InputFullDate, InputTime, InputSelect } from '../Inputs';
+import { CustomButton } from '../Button/CustomButton';
+import { useNavigate } from 'react-router-dom';
+
+import { Grid } from '@mui/material';
+
+import { db, auth } from '../../config/firebase';
+import { paths } from '../../config/paths';
+
+export const MyVisitForm = () => {
+  const [loading, setLoading] = useState(true);
+  const [pets, setPets] = useState([]);
+  const [pet, setPet] = useState('');
+  const [users, setUsers] = useState([]);
+  const [doctor, setDoctor] = useState('');
+  const [date, setDate] = useState(new Date());
+  const [hour, setHour] = useState(new Date());
+  const [doctorId, setDoctorId] = useState('');
+  const [doctorsNames, setDoctorsNames] = useState([]);
+  const [userName, setUserName] = useState('');
+  const user = auth.currentUser;
+  const userId = auth.currentUser.auth.lastNotifiedUid;
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const getPetsFromFirebase = [];
+    const getUsersFromFirebase = [];
+    db.collection('users').onSnapshot((querySnapshot) => {
+      querySnapshot.forEach((doc) => {
+        getUsersFromFirebase.push({
+          ...doc.data(),
+          key: doc.id,
+        });
+      });
+      setUsers(getUsersFromFirebase);
+      setLoading(false);
+    });
+    db.collection('pets').onSnapshot((querySnapshot) => {
+      querySnapshot.forEach((doc) => {
+        getPetsFromFirebase.push({
+          ...doc.data(),
+          key: doc.id,
+        });
+      });
+      setPets(getPetsFromFirebase);
+      setLoading(false);
+    });
+  }, [loading]);
+
+  const petsNames = [];
+  pets.forEach((pet) => {
+    if (pet.user_id === userId) {
+      petsNames.push(pet.name);
+    }
+  });
+
+  useEffect(() => {
+    let petOwner = '';
+    users.forEach((user) => {
+      if (user.uid === userId) {
+        petOwner = user;
+        return petOwner;
+      }
+    });
+    const petOwnerName = `${petOwner.firstName} ${petOwner.lastName}`;
+    setUserName(petOwnerName);
+  }, [userId, users]);
+
+  useEffect(() => {
+    const doctorsNames = [];
+    const doctorsArray = [];
+    let selectedDoctor = '';
+
+    users.forEach((doctor) => {
+      if (doctor.isAdmin) {
+        doctorsArray.push(doctor);
+        const doctorName = `${doctor.firstName} ${doctor.lastName}`;
+        doctorsNames.push(doctorName);
+        return doctorsNames;
+      }
+    });
+    setDoctorsNames(doctorsNames);
+
+    doctorsArray.forEach((el) => {
+      const name = `${el.firstName} ${el.lastName}`;
+      if (name === doctor) {
+        selectedDoctor = el.uid;
+        return selectedDoctor;
+      }
+    });
+    setDoctorId(selectedDoctor);
+  }, [doctor, users]);
+
+  if (loading) {
+    return <h1>loading data...</h1>;
+  }
```

Comment:
these lines of code shoud be lowerr, directly above main return

Reply:
fixed